src/msw/headerctrl.cpp

```diff
@@ -874,7 +1000,167 @@ bool wxHeaderCtrl::MSWOnNotify(int idCtrl, WXLPARAM lParam, WXLPARAM *result)
+}
+
+bool wxHeaderCtrl::Create(wxWindow *parent,
+                          wxWindowID id,
+                          const wxPoint& pos,
+                          const wxSize& size,
+                          long style,
+                          const wxString& name)
+{
+    long newStyle = style | wxCLIP_CHILDREN | wxTAB_TRAVERSAL;
+    if ( !wxWindow::Create(parent, id, pos, size, newStyle, name) )
+        return false;
+
+    m_nativeControl = new wxMSWHeaderCtrl(*this);
+    if ( !m_nativeControl->Create(this,
+                                  wxID_ANY,
+                                  wxDefaultPosition,
+                                  wxDefaultSize,
+                                  ApplyHeaderReorderFlagToStyle(wxNO_BORDER),
+                                  wxMSWHeaderCtrlNameStr) )
+        return false;
+
+    Bind(wxEVT_SIZE, &wxHeaderCtrl::OnSize, this);
+
+    return true;
+}
+
+wxHeaderCtrl::~wxHeaderCtrl()
+{
+    delete m_nativeControl;
+}
+
+// ----------------------------------------------------------------------------
+// wxHeaderCtrl events
+// ----------------------------------------------------------------------------
+
+void wxHeaderCtrl::OnSize(wxSizeEvent& WXUNUSED(event))
+{
+    if (m_nativeControl != NULL) // check whether initialisation has been done
+    {
+        int cw, ch;
+        GetClientSize(&cw, &ch);
+
+        m_nativeControl->SetSize(0, 0, cw, ch);
+    }
+}
+
+// ----------------------------------------------------------------------------
+// wxHeaderCtrl scrolling
+// ----------------------------------------------------------------------------
+
+void wxHeaderCtrl::DoScrollHorz(int dx)
+{
+    m_nativeControl->ScrollHorz(dx);
+}
+
+// ----------------------------------------------------------------------------
+// wxHeaderCtrl geometry calculation
+// ----------------------------------------------------------------------------
+
+wxSize wxHeaderCtrl::DoGetBestSize() const
+{
+    return m_nativeControl->GetBestSize();
+}
+
+// ----------------------------------------------------------------------------
+// wxHeaderCtrl columns managements
+// ----------------------------------------------------------------------------
+
+unsigned int wxHeaderCtrl::DoGetCount() const
+{
+    return m_nativeControl->GetCount();
+}
+
+void wxHeaderCtrl::DoSetCount(unsigned int count)
+{
+    m_nativeControl->SetCount(count);
+}
+
+void wxHeaderCtrl::DoUpdate(unsigned int idx)
+{
+    m_nativeControl->Update(idx);
+}
+
+void wxHeaderCtrl::DoSetColumnsOrder(const wxArrayInt& order)
+{
+    m_nativeControl->SetColumnsOrder(order);
+}
+
+wxArrayInt wxHeaderCtrl::DoGetColumnsOrder() const
+{
+    return m_nativeControl->GetColumnsOrder();
+}
+
+// ----------------------------------------------------------------------------
+// wxHeaderCtrl appearance
+// ----------------------------------------------------------------------------
+
+bool wxHeaderCtrl::SetBackgroundColour(const wxColour& colour)
```

Comment:
We could get rid of these forwarders by deriving `wxHeaderCtrl` from `wxCompositeWindowSettersOnly` and I think it would be worth doing this.